src/routes/Profile.jsx

```diff
@@ -0,0 +1,131 @@
+// React Imports
+import React, { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+// Inrupt Imports
+import { useSession } from '@inrupt/solid-ui-react';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Link from '@mui/material/Link';
+import Typography from '@mui/material/Typography';
+// Utility Imports
+import { fetchProfileInfo, updateProfileInfo } from '../utils';
+// Component Inputs
+import ProfileInputField from '../components/Profile/ProfileInputField';
+
+/**
+ * Profile Page - Page that displays the user's profile card information and
+ * allow users to edit/update them on PASS
+ *
+ * @memberof Pages
+ * @name Profile
+ * @returns {React.JSX.Element} The Profile Page
+ */
+const Profile = () => {
+  const location = useLocation();
+  const { session } = useSession();
+
+  localStorage.setItem('restorePath', location.pathname);
+
+  const [profileName, setProfileName] = useState(null);
+  const [memberOf, setMemberOf] = useState(null);
+  const [editProfileName, setEditProfileName] = useState(false);
```

Comment:
I feel like this state could be local to the `ProfileInputField`. Then you wouldn't need to pass it in or define a bunch of booleans up here.

Reply:
Moving edit states into `ProfileInputField`. Also moved form into each child component to more easily handle form submission and edit state.